eth_common/eth_types.nim

```diff
@@ -121,14 +121,33 @@ else:
 # Rlp serialization:
 #
 
+func trailingNonZeros(val: openarray[byte]): int =
+  for i in countdown(val.len - 1, 0):
+    if val[i] == 0:
+      return val.len - i - 1
+
+  return val.len
+
 proc read*(rlp: var Rlp, T: typedesc[Stint|StUint]): T {.inline.} =
   if rlp.isBlob:
-    result = readUintBE[result.bits](rlp.toBytes.toOpenArray)
+    let bytes = rlp.toBytes
+    if bytes.len > 0:
+      result = readUintBE[result.bits](bytes.toOpenArray)
+    else:
```

Comment:
Shouldn't there be a special case for value < 128? I'd like a test to make sure serialization/deserializing works properly.

Reply:
The special case is covered by the `isBlob` check. See the use of `rlp.getByteValue.to(T)` below.